guide/glossary.md

```diff
@@ -160,36 +183,70 @@ The path to the first address in a bitcoin-wallet using BIP84 will look like thi
 
-### YPUB
+### Keys
+Bitcoin uses public-key cryptography. Public-key cryptography uses randomly generated pair of keys comprising of a [public](#public-key) (which may be known to others) and [private key](#private-keys) (which should be kept secret from anyone but the owner).
 
-Same as XPUB however the Y denotes that this xpub belongs to a wallet that is following the [BIP49](https://github.com/bitcoin/bips/blob/master/bip-0049.mediawiki) standard that details the derivation scheme for wrapped-segwit addresses (P2WPKH-nested-in-P2SH).
+**[Private key](https://en.bitcoin.it/wiki/Private_key)**<br/>
 
-### ZPUB
+Every bitcoin address has a public key and a corresponding private key, together they are called a keypair. If you have access to both the public and private key, you effectively control the funds in the address. As with HD Wallets there are also keypairs that control *branches* in the hierarchical tree of the wallet, and at the very top is the extended keypair (x-pub and x-prv for short) that control all the addresses in the wallet.
 
-Same as YPUB though the Z denotes it is an extended public key from a segregated witness enabled wallet following [BIP84](https://github.com/bitcoin/bips/blob/master/bip-0084.mediawiki).
+The private key is a 64 hexadecimal (or 256 if described in binary 1’s and 0’s) character string generated by the encryption algorithm. They look something like this in hexadecimal form:
 
-## Initial block download (IBD)
+`1E99423A4ED27608A15A2616A2B0E9E52CED330AC530EDCC32C8FFC6A526AEDD`
 
-To fully verify that all transactions on the bitcoin network are valid, a full node needs to download and examine all previous block-data. This is called the initial block download, after which the node has caught up with the latest transaction activity. This can take several hours, and only after it is complete can wallets linked to the node be used.
+**Extended private key (xpriv)**<br/>
 
-## Input
+In hierarchical deterministic (HD) wallets, all a wallets private keys are derived from this XPRIV.
+
+`xprv9zrji5mK3nb4RbuR2ZYFtyzK3gn78KnEzkNP4ZxwwPPwcgQQVZqnjTMAGxmmM3jpmfsthQUtfD9iYPvnaqwejCjcyEswLqEhX4LPKNFUXT5`
+
+**Public key**<br/>
+
+Bitcoin public keys are derived from private keys. Public keys can not derive private keys however. A Bitcoin addresses(#address) is generated by hashing a public key.
+
+`02f062b2649e0665f2e8e5573172ef85921fc4b307e9795113930ed2229d271f4e`
+
+**Extended public key (xpub)**<br/>
+
+In hierarchical deterministic (HD) wallets, all addresses and their matching public keys are derived from this XPRIV. All public keys and subsequent addresses (by hashing these public keys) are generated from it.
+
+`xpub6DE1bvvTaMkDaAghedChtufg3rDPeYdWt9sM5iTwBVYe9X6bmLenQrSexSa1qDscYtidSMUEo9u7TuXg48Y3hBXQr7Yw8zUaEToH1rVgvif`
+
+**ypub** - A xpub belonging to a wallet deriving P2WPKH in P2SH addresses.
```

Comment:
I find this super hard to parse (for example, `deriving P2WSH in P2SH addresses`). And not sure what I would do with this as a designer. Is this the point where we keep it simple and refer to external, more technical documentation? Could we just say that there are variations of xpubs created by different projects and it's important to notify users about any potential issues they need to be aware of... or something like that?

Reply:
I was (or maybe someone else can) elaborate on these further. For now I just put that in as a placeholder. I guess including it was merely to make designers aware that these xpub derivatives exist and are case sensitive.